src/engine_registry.cpp

```diff
@@ -75,331 +75,329 @@ namespace wr
+		.entry = "main_cs",
+		.type = ShaderType::DIRECT_COMPUTE_SHADER
+	});
 
 	REGISTER(shaders::mip_mapping_cs, ShaderRegistry)({
-		ShaderDescription::Path("resources/shaders/generate_mips_cs.hlsl"),
-		ShaderDescription::Entry("main"),
-		ShaderDescription::Type(ShaderType::DIRECT_COMPUTE_SHADER)
-		});
+		.path = "resources/shaders/generate_mips_cs.hlsl",
+		.entry = "main",
+		.type = ShaderType::DIRECT_COMPUTE_SHADER
+	});
 
 	REGISTER(shaders::equirect_to_cubemap_vs, ShaderRegistry)({
-		ShaderDescription::Path("resources/shaders/equirect_to_cubemap_conversion.hlsl"),
-		ShaderDescription::Entry("main_vs"),
-		ShaderDescription::Type(ShaderType::VERTEX_SHADER)
-		});
+		.path = "resources/shaders/equirect_to_cubemap_conversion.hlsl",
+		.entry = "main_vs",
+		.type = ShaderType::VERTEX_SHADER
+	});
 
 	REGISTER(shaders::equirect_to_cubemap_ps, ShaderRegistry)({
-		ShaderDescription::Path("resources/shaders/equirect_to_cubemap_conversion.hlsl"),
-		ShaderDescription::Entry("main_ps"),
-		ShaderDescription::Type(ShaderType::PIXEL_SHADER)
-		});
+		.path = "resources/shaders/equirect_to_cubemap_conversion.hlsl",
+		.entry = "main_ps",
+		.type = ShaderType::PIXEL_SHADER
+	});
 
 	REGISTER(shaders::cubemap_convolution_ps, ShaderRegistry)({
-		ShaderDescription::Path("resources/shaders/cubemap_convolution.hlsl"),
-		ShaderDescription::Entry("main_ps"),
-		ShaderDescription::Type(ShaderType::PIXEL_SHADER)
-		});
+		.path = "resources/shaders/cubemap_convolution.hlsl",
+		.entry = "main_ps",
+		.type = ShaderType::PIXEL_SHADER
+	});
 
 	REGISTER(shaders::cubemap_prefiltering_cs, ShaderRegistry)({
-	ShaderDescription::Path("resources/shaders/prefilter_env_map_cs.hlsl"),
-	ShaderDescription::Entry("main_cs"),
-	ShaderDescription::Type(ShaderType::DIRECT_COMPUTE_SHADER)
-		});
+		.path = "resources/shaders/prefilter_env_map_cs.hlsl",
+		.entry = "main_cs",
+		.type = ShaderType::DIRECT_COMPUTE_SHADER
+	});
 
 	REGISTER(pipelines::brdf_lut_precalculation, PipelineRegistry) < Vertex2D > ({
-		PipelineDescription::VertexShader(std::nullopt),
-		PipelineDescription::PixelShader(std::nullopt),
-		PipelineDescription::ComputeShader(shaders::brdf_lut_cs),
-		PipelineDescription::RootSignature(root_signatures::brdf_lut),
-		PipelineDescription::DSVFormat(Format::UNKNOWN),
-		PipelineDescription::RTVFormats({ Format::R16G16_FLOAT }),
-		PipelineDescription::NumRTVFormats(1),
-		PipelineDescription::Type(PipelineType::COMPUTE_PIPELINE),
-		PipelineDescription::CullMode(CullMode::CULL_BACK),
-		PipelineDescription::Depth(false),
-		PipelineDescription::CounterClockwise(true),
-		PipelineDescription::TopologyType(TopologyType::TRIANGLE)
-		}
-	);
+		.m_vertex_shader_handle = std::nullopt,
+		.m_pixel_shader_handle = std::nullopt,
+		.m_compute_shader_handle = shaders::brdf_lut_cs,
+		.m_root_signature_handle = root_signatures::brdf_lut,
+		.m_dsv_format = Format::UNKNOWN,
+		.m_rtv_formats = { Format::R16G16_FLOAT },
+		.m_num_rtv_formats = 1,
```

Comment:
m_num_rtv_formats is a uint, might be a good idea to explicitly assign an unsigned value. Not too important though, just consistancy as we do this for most of the project.